Mathlib/CategoryTheory/Limits/Shapes/Kernels.lean

```diff
@@ -223,6 +223,12 @@ def KernelFork.isLimitOfIsLimitOfIff' {X Y : C} {g : X ⟶ Y} {c : KernelFork g}
   IsLimit.ofIsoLimit (isLimitOfIsLimitOfIff hc g' (Iso.refl _) (by simpa using iff))
     (Fork.ext (Iso.refl _))
 
+lemma KernelFork.IsLimit.isZero_of_mono {X Y : C} {f : X ⟶ Y}
+    {c : KernelFork f} (hc : IsLimit c) [Mono f] : IsZero c.pt := by
+  have := Fork.IsLimit.mono hc
+  rw [IsZero.iff_id_eq_zero, ← cancel_mono c.ι, ← cancel_mono f, Category.assoc,
+    Category.assoc, c.condition, comp_zero, zero_comp]
```

Comment:
It is surprising that we didn't have this until now. Can you add the iff version using `mono_of_isZero_kernel'` (Is it really only true for preadditive categories?)? Thanks.

Reply:
Thanks for the suggestion! I have added these lemmas in https://github.com/leanprover-community/mathlib4/pull/22739/commits/db66f764d98986b6860c02a0d5a88722875de1e6